seatunnel-connectors/seatunnel-connector-spark-clickhouse/src/main/scala/org/apache/seatunnel/spark/sink/Clickhouse.scala

```diff
@@ -222,8 +222,22 @@ class Clickhouse extends SparkBatchSink {
         statement.setInt(index + 1, item.getAs[Int](fieldIndex))
       case "UInt32" | "UInt64" | "Int64" =>
         statement.setLong(index + 1, item.getAs[Long](fieldIndex))
-      case "Float32" => statement.setFloat(index + 1, item.getAs[Float](fieldIndex))
-      case "Float64" => statement.setDouble(index + 1, item.getAs[Double](fieldIndex))
+      case "Float32" =>
+        val value = item.get(fieldIndex)
+        value match {
+          case decimal: BigDecimal =>
+            statement.setFloat(index + 1, decimal.floatValue())
+          case _ =>
+            statement.setFloat(index + 1, value.asInstanceOf[Float])
+        }
```

Comment:
@BenJFan `Clickhouse` seems to have a `Decimal` data type, haven't tried to use this?

Reply:
This problem is sink to clickhouse user want the data type is double/float. But ours data type is decimal

Reply:
> This problem is sink to clickhouse user want the data type is double/float. But ours data type is decimal

Okay, see, can we add `decimal` too?

Reply:
> 

OK, I will do it

Reply:
> > This problem is sink to clickhouse user want the data type is double/float. But ours data type is decimal
> 
> Okay, see, can we add `decimal` too?

sorry, I find clickhouse sink already have decimal. See Clickhouse.scala:243